vdev/src/git.rs

```diff
@@ -63,6 +63,62 @@ pub fn get_modified_files() -> Result<Vec<String>> {
     Ok(capture_output(&args)?.lines().map(str::to_owned).collect())
 }
 
+pub fn set_config_values(config_values: Vec<(&str, &str)>) -> Result<String> {
+    let mut args = Vec::new();
+    args.push("config");
+    // args.push("--global");
+
+    for (key, value) in config_values {
+        args.push(key);
+        args.push(value);
+    }
+
+    capture_output(&args)
+}
+
+// Checks if the current directory's repo is clean
+pub fn check_git_repository_clean() -> Result<bool> {
+    let output = Command::new("git")
+        .arg("diff-index")
+        .arg("--quiet")
+        .arg("HEAD")
+        .output()
+        .map_err(|e| anyhow!("{}", e))?;
+
+    Ok(output.status.success())
+}
+
+// Commits changes from the current repo
+pub fn commit(commit_message: &str) -> Result<Output> {
+    let output = Command::new("git")
+        .arg("-am")
+        .arg(commit_message)
+        .output()
+        .map_err(|e| anyhow!("{}", e))?;
+
+    Ok(output)
+}
```

Comment:
Hmm, thinking a little further, I don't think these are going to return an error to the caller if the underlying command fails, only if spawning the command and catching the result fails. I think we need a different wrapper for these kinds of commands where we don't care about the output but do need to catch the exit code as an error.

Reply:
We previously discussed modifying the capture_output() function, but for now, I've implemented a new wrapper function called run_command_check_status(). This function runs the command, checks its exit status for success, returns an error if either the spawning of the command fails or if the command's exit code indicates a failure. This does so without caring about the output.

Reply:
Tracking here: https://github.com/vectordotdev/vector/issues/16807